hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -336,9 +336,10 @@ private void ensurePartitionLoadedCorrectly(String partition) {
       if (!metaClient.getFs().exists(partitionPath)) {
         metaClient.getFs().mkdirs(partitionPath);
         return new FileStatus[0];
+      } else {
+        return metaClient.getFs().listStatus(partitionPath);
```

Comment:
Can we add one line comment here: 
```java
// in case the partition path was created by another caller.
```

Reply:
+1

Reply:
in general, I thought list partitions for any given partition can't have concurrent callers. across partitions we will parallelize, but for a given partition, I wasn't aware that there could be concurrent calls. can you throw some light on when this could happen.

Reply:
In flink pipeline, there are cases that we need to look into the partition files through the fs view with this method call, the bucket assigner would check the small files under the partition and the partition may be new.

Multiple bucket assigner may operate on same partitions at the same time.